### PR TITLE
Add Podzol Rooting Soil

### PR DESCRIPTION
**Pull request by Bloo-dev**
Allows Sapling to self-plant when on Podzol.
This module was carefully crafted to only do `@e[type=item]` every 15 seconds and respects the `randomTickSpeed` Gamerule.
All types of trees are supported.

saplings-on-podzol

### CODE REVIEW

File: gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/main.mcfunction [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# function loops on a 1-minute (1200t or 0.01666Hz) cycle
+schedule function gm4_podzol_rooting_soil:main 300t
```
Comment (BluePsychoRanger) [inline gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/main.mcfunction]:
```suggestion
# function loops on a 30-second (600t or 0.5Hz) cycle
schedule function gm4_podzol_rooting_soil:main 600t
```

File: gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/randomizer_loop.mcfunction [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+execute store result score current_uuid gm4_podzol_data run data get entity @s UUIDMost 0.00000000023283064365386962890625
+scoreboard players operation current_uuid gm4_podzol_data %= #100 gm4_podzol_data
+
+# copy into target and shift by random_tick_speed
+scoreboard players operation compared_value gm4_podzol_data = current_uuid gm4_podzol_data
+scoreboard players operation compared_value gm4_podzol_data *= random_tick_speed gm4_podzol_data
+
+# compare
+execute if score compared_value gm4_podzol_data >= last_uuid gm4_podzol_data if entity @s[nbt={Item:{Count:1b}}] at @s if block ~ ~-1 ~ minecraft:podzol if block ~ ~ ~ #gm4_podzol_rooting_soil:air run function gm4_podzol_rooting_soil:plant
+
```
Comment (BluePsychoRanger) [inline gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/randomizer_loop.mcfunction]:
```suggestion

#force plant if about to despawn
execute store result score age gm4_podzol_data run data get entity @s Age
execute if score age gm4_podzol_data matches 5399.. if entity @s[nbt={Item:{Count:1b}}] at @s if block ~ ~-1 ~ minecraft:podzol if block ~ ~ ~ #gm4_podzol_rooting_soil:air run function gm4_podzol_rooting_soil:plant

```

File: gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/plant.mcfunction [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# @s = sapling item
+# at @s
+# called from gm4_podzol_rooting_soil:randomizer_uuid
```
Comment (BluePsychoRanger) [inline gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/plant.mcfunction]:
```suggestion
# called from gm4_podzol_rooting_soil:randomizer_loop
```

File: gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/main.mcfunction (hunk 1 above)
Comment (Kroppeb) [reply gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/main.mcfunction]: ... 0.033333 He
0.5 is every 2 seconds

File: gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/plant/force.mcfunction [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# @s = sapling item on podzol and within air-like block that has failed the randomizer
+# at world spawn
+# called from gm4_podzol_rooting_soil:randomizer
+
+execute store result score age gm4_podzol_data run data get entity @s Age
+execute if score age gm4_podzol_data matches 5399.. at @s if block ~ ~-1 ~ minecraft:podzol if block ~ ~ ~ #gm4_podzol_rooting_soil:air run function gm4_podzol_rooting_soil:plant
```
Comment (BluePsychoRanger) [inline gm4_podzol_rooting_soil/data/gm4_podzol_rooting_soil/functions/plant/force.mcfunction]:
```suggestion
execute if score age gm4_podzol_data matches 5399.. at @s if block ~ ~-1 ~ minecraft:podzol if block ~ ~ ~ #gm4_podzol_rooting_soil:air run function gm4_podzol_rooting_soil:plant/normal
```

**Pull request merged**